app/assets/javascripts/discourse/app/components/dialog-messages/confirm-session.gjs

```diff
@@ -0,0 +1,95 @@
+import Component from "@glimmer/component";
```

Comment:
This is very similar to the password screen we show on the 2FA screen, however the reason why it's a dialog is so that all login tools (password reset, passkeys, 2FAs, etc.) can eventually be on the same screen. 

<img width="697" alt="image" src="https://github.com/discourse/discourse/assets/368961/57e4b176-1804-439e-aba0-74ff2a91f4ac">

In a future PR, I think it would make sene to change the location of the 2FA items and place them in the same screen, under `/u/username/preferences/security` (as opposed to `u/username/preferences/second-factor` which also doesn't have a correct active menu item highlighted).